repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -166,10 +168,31 @@ def prepare_target_userspace(context, userspace_dir, enabled_repos, packages):
         try:
             context.call(cmd, callback_raw=utils.logging_handler)
         except CalledProcessError as exc:
-            raise StopActorExecutionError(
-                message='Unable to install RHEL {} userspace packages.'.format(target_major_version),
-                details={'details': str(exc), 'stderr': exc.stderr}
-            )
+            message = 'Unable to install RHEL {} userspace packages.'.format(target_major_version)
+            details = {'details': str(exc), 'stderr': exc.stderr}
+
+            # If a proxy was set in dnf config, it should be the reason why dnf
+            # failed since leapp does not support updates behind proxy yet.
+            for manager_info in api.consume(PkgManagerInfo):
+                if manager_info.configured_proxies:
+                    raise StopActorExecutionError(
+                        message=message,
+                        details=("DNF failed to install userspace packages, likely due to the proxy "
+                                 "configuration detected in the YUM/DNF configuration file.")
+                    )
+
+            # Similarly if a proxy was set specifically for one of the repositories.
+            for repo_facts in api.consume(RepositoriesFacts):
+                for repo_file in repo_facts.repositories:
+                    for repo_data in repo_file.data:
```

Comment:
Nit: you could remove at least one cycle if you use `any`, like

```
for repo_facts in api.consume(RepositoriesFacts):
  for repo_file in repo_facts.repositories:
    if any(repo_data.proxy for repo_data in repo_file.data):
       raise StopActorExecutionError(...)
```

Reply:
Thanks, that's a useful tip!